src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -48,7 +58,11 @@ public String findElement(List<String> elements, String element) {
      * NoSuchElementException</p>
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+        return IntStream.range(0, numbers.size())
+                .map(index -> (index & 1) == 1 ? numbers.get(index) - 1 : numbers.get(index))
+                .filter(number -> (number & 1) == 1)
```

Comment:
I hope you understand what does it mean)

Reply:
Yep, I do :)